test/test-bindings-characteristic.js

```diff
@@ -0,0 +1,79 @@
+
+var should = require('should');
+var sinon = require('sinon');
+var bindings = require('../lib/mac/bindings');
+var Mock = require('./core-bluetooth-mock');
+
+var a = require('./common');
+
+describe('Noble bindings characteristic', function() {
+  var sandbox;
+  var mock;
+
+  beforeEach(function() {
+    sandbox = sinon.sandbox.create();
+    mock = new Mock(bindings, sandbox);
+
+    //set internal state
+    mock.discoverPeripheral();
+    mock.discoverServices();
+    mock.discoverCharacteristics();
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+    mock = null;
+  });
+
+  it('should emit read', function() {
+    var eventSpy = sandbox.spy();
+    bindings.once('read', eventSpy);
+
+    //stub native
+    sandbox.stub(mock.nativeCharacteristicObject, "readValue", function(){
+      this.emit('valueUpdate', a.dataBuffer);
+    });
+
```

Comment:
Were you thinking we could create test helpers?

For example, tests for another bindings might have a very similar setup to this, so maybe there's an opportunity to share some test code between them?

You could pass in the binding, maybe have hook to setup stubs and another for verify ...